### PR TITLE
Channel deletion

### PR DESCRIPTION
**Pull request by siepra**

### CODE REVIEW

File: packages/backend/src/storage/ChannelsAccessController.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export const createChannelAccessController = (peerId) => {
+  // @ts-ignore
+  class ChannelsAccessController extends AccessController {
+    static get type() {
+      return type
+    }
+
+    async canAppend(entry, identityProvider) {
+      // Channel deletion WIP
```
Comment (siepra) [inline packages/backend/src/storage/ChannelsAccessController.ts:17]: Is this still WIP?

File: packages/desktop/libssl.so
Comment (siepra) [inline packages/desktop/libssl.so:1]: Are we sure we don't need it anymore? What about Ubuntu 22.04?

File: packages/desktop/package.json [hunk 2]
```diff
@@ -80,7 +80,7 @@
   },
   "homepage": "https://github.com/TryQuiet",
   "@comment version": "To build new version for specific platform, just replace platform in version tag to one of following linux, mac, windows",
-  "version": "1.2.0",
+  "version": "1.2.0-alpha.31",
```
Comment (siepra) [inline packages/desktop/package.json:83]: Isn't this version behind the full 1.2.0?

File: packages/state-manager/src/sagas/messages/messages.slice.ts [hunk 3]
```diff
@@ -38,6 +39,10 @@ export const messagesSlice = createSlice({
   name: StoreKeys.Messages,
   reducers: {
     sendMessage: (state, _action: PayloadAction<WriteMessagePayload>) => state,
+    deleteMessages: (state, action: PayloadAction<DeleteMessagesPayload>) => {
```
Comment (siepra) [inline packages/state-manager/src/sagas/messages/messages.slice.ts:42]: I think this name may be confusing. It sounds like it deletes certain amount of messages, How about marking that we delete the whole channel entry for messages?

File: packages/state-manager/src/sagas/publicChannels/deletedChannel/deletedChannel.saga.test.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            description: 'Welcome to #photo',
+            timestamp: DateTime.utc().valueOf(),
+            owner: owner.nickname,
+            address: 'photo'
+          }
+        }
+      )
+    ).channel
+  })
+  describe('owner handling', () => {
```
Comment (siepra) [inline packages/state-manager/src/sagas/publicChannels/deletedChannel/deletedChannel.saga.test.ts:60]: I like test names to be more descriptive. If I saw 'owner handling' test failing, I wouldn't know what exactly doesn't work

File: packages/state-manager/src/sagas/publicChannels/publicChannels.slice.ts [hunk 5]
```diff
@@ -29,24 +33,44 @@ const log = logger('publicChannels')
 
 export class PublicChannelsState {
   public currentChannelAddress: string = 'general'
+  public isGeneralRecreation: boolean = false
```
Comment (siepra) [inline packages/state-manager/src/sagas/publicChannels/publicChannels.slice.ts:36]: Once again it's just a minor naming thing, but 'pendingGeneralChannelRecreation' would be more accurate, in my opinion

File: packages/desktop/libssl.so
Comment (vinkabuki) [reply packages/desktop/libssl.so:1]: It was added here by accident

File: packages/desktop/package.json (hunk 2 above)
Comment (vinkabuki) [reply packages/desktop/package.json:83]: right

File: packages/state-manager/src/sagas/publicChannels/publicChannels.slice.ts (hunk 5 above)
Comment (vinkabuki) [reply packages/state-manager/src/sagas/publicChannels/publicChannels.slice.ts:36]: +1

File: packages/state-manager/src/sagas/publicChannels/deletedChannel/deletedChannel.saga.ts
[the remark this answers is not in the document]
Comment (Kacper-RF) [reply packages/state-manager/src/sagas/publicChannels/deletedChannel/deletedChannel.saga.ts:1]: I will do that in next PR

**Review by vinkabuki — APPROVED** [review]

**Review by vinkabuki — APPROVED** [review]

File: packages/backend/src/storage/ChannelsAccessController.ts (hunk 1 above)
Comment (holmesworcester) [reply packages/backend/src/storage/ChannelsAccessController.ts:17]: We merged this just so that emi's typescript work could move forward?

**Pull request merged**